helper/reports.go

```diff
@@ -0,0 +1,76 @@
+package helper
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+)
+
+func ProcessDownloadReports(fileID int, alwaysApproved bool, generate string, output string) (bool, error) {
+	var resultID int
+
+	fmt.Println("Warning: This process will download report file to system.")
+	if !alwaysApproved {
+		fmt.Println("Please pass `--always-approved` to approve all the reports")
+		return false, errors.New("Please pass `--always-approved` to approve all the reports")
+	}
+
+	ctx := context.Background()
+	client := getClient()
+
+	if generate == "yes" {
+		// This part of code is to generate reports
+		fmt.Println("Generating reports...")
+		result, err := client.Reports.GenerateReport(ctx, fileID)
+		if err != nil {
+			PrintError(err)
+			return false, err
+		}
+
+		if result != nil && result.ID == 0 {
+			err := errors.New("A report is already being generated or scan is in progress. Please wait.")
```

Comment:
I think this code would be unreachable. The `client.Reports.GenerateReport` will set `err` to `non-nil` value, specifically setting it to the content of error response if the API returns an error. The result/response will be nil in that case. Correct me If I'm wrong.

Reply:
In this case it is sending status code as bad request. But api client doesn't sending error. It sending body that sends from api server.

Reply:
fixed